src/org/zaproxy/zap/extension/tlsdebug/HttpsCallerProcess.java

```diff
@@ -0,0 +1,69 @@
+package org.zaproxy.zap.extension.tlsdebug;
+
+import static org.junit.Assert.*;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.protocol.Protocol;
+import org.apache.commons.httpclient.protocol.ProtocolSocketFactory;
+import org.junit.Assert;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.network.SSLConnector;
+
+import ch.csnc.extension.httpclient.SSLContextManager;
+
+public class HttpsCallerProcess {
+
+    private static final String HTTPS = "https";
+
+    public static void main(String[] args) {

```

Comment:
Purpose would be to use the configuration of ZAP as much as possible, in particular the client certificates, but any relevant configuration as well (e.g. the aforementioned outgoing proxy, enabled TLS/SSL protocold). But I need the help and advice of ZAP developers for that.